examples/pcmopus/subscribing/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	webm "github.com/livekit/mediatransportutil/pkg/audio/webm"
+	"github.com/livekit/protocol/logger"
+	lksdk "github.com/livekit/server-sdk-go/v2"
+	"github.com/pion/webrtc/v4"
+)
+
+const (
+	host                = "ws://localhost:7880"
+	apiKey              = "devkey"
+	apiSecret           = "secret"
+	roomName            = "test"
+	participantIdentity = "go-sdk-subscriber"
+)
+
+func main() {
+	logger.InitFromConfig(&logger.Config{Level: "info"}, "pcmopus")
+	lksdk.SetLogger(logger.GetLogger())
+
+	var pcmTrack *lksdk.OpusToPCM16AudioTrack
+	var fileWriter *os.File
+
+	room, err := lksdk.ConnectToRoom(host, lksdk.ConnectInfo{
+		APIKey:              apiKey,
+		APISecret:           apiSecret,
+		RoomName:            roomName,
+		ParticipantIdentity: participantIdentity,
+	}, &lksdk.RoomCallback{
+		ParticipantCallback: lksdk.ParticipantCallback{
+			OnTrackSubscribed: func(track *webrtc.TrackRemote, publication *lksdk.RemoteTrackPublication, rp *lksdk.RemoteParticipant) {
+				pcmTrack, fileWriter = onTrackSubscribed(track, publication)
```

Comment:
This should:
- Use a mutex, we don't want SDK user to complain about subtle bugs caused by races.
- Check if the track is already assigned. Even for 1:1 there might be more than one track, potentially.
- Ignore video tracks.

I think most of the problems can be fixed by just using an unbuffered channel and select-default to send to it.

Reply:
Ignored non-opus tracks, do we need the rest? This is only an example

Reply:
Right, but users will likely copy-paste it, so having a mutex/channel at least is a good idea, I think.